spec.html

```diff
@@ -35566,7 +35582,7 @@ <h1>
         </emu-alg>
         <emu-grammar>AtomEscape :: `k` GroupName</emu-grammar>
         <emu-alg>
-          1. Search the enclosing |Pattern| for an instance of a |GroupSpecifier| containing a |RegExpIdentifierName| which has a CapturingGroupName equal to the CapturingGroupName of the |RegExpIdentifierName| contained in |GroupName|.
+          1. Search the enclosing |Pattern| for an instance of a |GroupSpecifier| containing a |RegExpIdentifierName| which has a CapturingGroupName equal to the CapturingGroupName of the |RegExpIdentifierName| contained in |GroupName|, and such that CanBothParticipate(|AtomEscape|, |GroupSpecifier|) is *true*.
           1. Assert: A unique such |GroupSpecifier| is found.
```

Comment:
This is actually wrong. Consider:

```js
/(?:(?<a>x)|(?<a>y))\k<a>/
```

This is legal as specified (and should be) - the duplicated name occurs in different alternative - but both of the `a` groups can participate alongside the backreference. So we no longer know statically precisely which group a backreference refers to.

This _should_ work - at most one of those two can have matched, so the backreference should be to whichever of the two did match - but these semantics don't get us there. Making this work will be more complicated.